app/scripts/controllers/permissions/loggerMiddleware.js

```diff
@@ -81,51 +81,50 @@ module.exports = function createLoggerMiddleware ({
     return Object.keys(request.params[0])
   }
 
-  function logHistory (requestedMethods, origin, result, time) {
-
-    let accounts
-    const entries = result
-      ? result.map(perm => {
-        if (perm.parentCapability === 'eth_accounts') {
-          accounts = getAccountsFromPermission(perm)
-        }
-        return perm.parentCapability
-      })
-        .reduce((acc, m) => {
-          if (requestedMethods.includes(m)) {
-            acc[m] = {
-              lastApproved: time,
-            }
-            if (m === 'eth_accounts') {
-              acc[m].accounts = accounts
-            }
+  function logHistory (requestedMethods, origin, result, time, isEthRequestAccounts) {
+    let accounts, entries
+    if (isEthRequestAccounts) {
+      accounts = result
+      entries = { 'eth_accounts': [ { account: accounts[0], lastApproved: time } ] }
+    } else {
+      entries = result
+        ? result.map(perm => {
+          if (perm.parentCapability === 'eth_accounts') {
+            accounts = getAccountsFromPermission(perm)
           }
-          return acc
-        }, {})
-      : {}
+          return perm.parentCapability
+        })
+          .reduce((acc, m) => {
+            if (requestedMethods.includes(m)) {
+              if (m === 'eth_accounts') {
+                acc[m] = accounts.map(account => ({ lastApproved: time, account }))
+              } else {
+                acc[m] = [ {
+                  lastApproved: time,
+                } ]
+              }
+            }
```

Comment:
Why are all entry values arrays? For instance, the point of `lastApproved` (at least the one directly associated with a permission) is to be a single record, per permission, of the last time the permission was approved. IMO, there is no need to store multiple version of it in the history log. More granular information is kept in the activity log. If we want to record _the first time_ it was exposed, I'd support adding a `firstApproved` property.

Regarding the `accounts` property of the `eth_accounts` permission, the point of that is to know which accounts have _ever_ been exposed to a specific domain. I suppose it can be valuable to the user to know the last time a specific account was exposed to a domain, so I support store accounts as `{ lastApproved, account }` in the `accounts` array. (I don't think adding `firstApproved` per account is helpful, but I don't feel strongly about it). For completeness, I think we should also record the last time `eth_accounts` was granted, period, without regard to specific accounts.

With that, here's an example of how the `entries` object should look for an arbitrary domain:
```javascript
abc.xyz.dapp: {

  eth_accounts: {
    lastApproved: `YYYY-mm-DD HH:MM`,
    accounts: [
      { lastApproved: `YYYY-mm-DD HH:MM`, account: '0xabc...' },
      { lastApproved: `YYYY-mm-DD HH:MM`, account: '0xdef...' },
      // ...
    ],
  },

  permission_x: {
    lastApproved: `YYYY-mm-DD HH:MM`,
  },
  // ...
}
```

Reply:
Reviewed changes to this, looks great.